components/style/values/specified/font.rs

```diff
@@ -851,6 +851,318 @@ impl Parse for FontVariantAlternates {
     }
 }
 
+bitflags! {
+    #[derive(MallocSizeOf)]
+    /// Vairants for east asian variant
+    pub struct VariantEastAsian: u16 {
+        /// None of the features
+        const NORMAL = 0;
+        /// Enables rendering of JIS78 forms (OpenType feature: jp78)
+        const JIS78 = 0x01;
+        /// Enables rendering of JIS83 forms (OpenType feature: jp83).
+        const JIS83 = 0x02;
+        /// Enables rendering of JIS90 forms (OpenType feature: jp90).
+        const JIS90 = 0x04;
+        /// Enables rendering of JIS2004 forms (OpenType feature: jp04).
+        const JIS04 = 0x08;
+        /// Enables rendering of simplified forms (OpenType feature: smpl).
+        const SIMPLIFIED = 0x10;
+        /// Enables rendering of traditional forms (OpenType feature: trad).
+        const TRADITIONAL = 0x20;
+        /// Enables rendering of full-width variants (OpenType feature: fwid).
+        const FULL_WIDTH = 0x40;
+        /// Enables rendering of proportionally-spaced variants (OpenType feature: pwid).
+        const PROPORTIONAL_WIDTH = 0x80;
+        /// Enables display of ruby variant glyphs (OpenType feature: ruby).
+        const RUBY = 0x100;
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl VariantEastAsian {
+    /// Obtain a specified value from a Gecko keyword value
+    ///
+    /// Intended for use with presentation attributes, not style structs
+    pub fn from_gecko_keyword(kw: u16) -> Self {
```

Comment:
Yeah, this is not great, the right way to do it is having the same bytes as the gecko keyword, and using `VariantEastAsian::from_bits_truncate` and `VariantEastAsiant::bits` to go from / to `u16`.

The cool part is, they match already I believe!

But we need to assert it so it doesn't break, so I'd add a `pub fn assert_variant_east_asian_matches()` similar to:

  https://github.com/servo/servo/blob/fe4139b779b3af749ec1426ddf4e1393c7b85442/components/style/invalidation/element/restyle_hints.rs#L238

And then make this function just:

```rust
Self::from_bits_truncate(kw)
```

And the `to_gecko_keyword` just:

```rust
self.bits()
```

Reply:
Updated!